Packer crash in EcrGetLogin

```
panic: runtime error: index out of range
2017/01/10 22:27:48 packer: 
2017/01/10 22:27:48 packer: goroutine 7 [running]:
2017/01/10 22:27:48 packer: panic(0x13beea0, 0xc42000e190)
2017/01/10 22:27:48 packer:     /usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1
2017/01/10 22:27:48 packer: github.com/mitchellh/packer/builder/docker.(*AwsAccessConfig).EcrGetLogin(0xc4202fe540, 0xc4202ef180, 0xa, 0x1, 0x1, 0xc4203acb70, 0x21, 0x4574f0, 0xc420015e40)
2017/01/10 22:27:48 packer:     /Users/mwhooker/go/src/github.com/mitchellh/packer/builder/docker/ecr_login.go:55 +0x8e6
2017/01/10 22:27:48 packer: github.com/mitchellh/packer/builder/docker.(*StepPull).Run(0x1e62d10, 0x1e117a0, 0xc42013c870, 0x0)
2017/01/10 22:27:48 packer:     /Users/mwhooker/go/src/github.com/mitchellh/packer/builder/docker/step_pull.go:27 +0x75a
2017/01/10 22:27:48 packer: github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc4203b02c0, 0x1e117a0, 0xc42013c870)
2017/01/10 22:27:48 packer:     /Users/mwhooker/go/src/github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x207
2017/01/10 22:27:48 packer: github.com/mitchellh/packer/builder/docker.(*Builder).Run(0xc420288120, 0x1e140e0, 0xc4203aa520, 0x1e0d0a0, 0xc4203a6e50, 0x1e12e60, 0xc4203100a8, 0xc42013c6c0, 0x0, 0x0, ...)
2017/01/10 22:27:48 packer:     /Users/mwhooker/go/src/github.com/mitchellh/packer/builder/docker/builder.go:82 +0x85e
2017/01/10 22:27:48 packer: github.com/mitchellh/packer/packer/rpc.(*BuilderServer).Run(0xc420288b40, 0x1, 0xc4200fc4e0, 0x0, 0x0)
2017/01/10 22:27:48 packer:     /Users/mwhooker/go/src/github.com/mitchellh/packer/packer/rpc/builder.go:93 +0x218
2017/01/10 22:27:48 packer: reflect.Value.call(0xc4202b2840, 0xc420282088, 0x13, 0x15ffd31, 0x4, 0xc4203d1ed0, 0x3, 0x3, 0x0, 0x0, ...)
2017/01/10 22:27:48 packer:     /usr/local/Cellar/go/1.7.3/libexec/src/reflect/value.go:434 +0x5c8
2017/01/10 22:27:48 packer: reflect.Value.Call(0xc4202b2840, 0xc420282088, 0x13, 0xc4203d1ed0, 0x3, 0x3, 0xc4202f5a98, 0x1, 0x0)
2017/01/10 22:27:48 packer:     /usr/local/Cellar/go/1.7.3/libexec/src/reflect/value.go:302 +0xa4
2017/01/10 22:27:48 packer: net/rpc.(*service).call(0xc42028a740, 0xc42028a700, 0xc4202ee010, 0xc4202b8700, 0xc4202f6280, 0x13117c0, 0xc4200fc4dc, 0x18a, 0x12e6d00, 0xc4200fc4e0, ...)
2017/01/10 22:27:48 packer:     /usr/local/Cellar/go/1.7.3/libexec/src/net/rpc/server.go:383 +0x148
2017/01/10 22:27:48 packer: created by net/rpc.(*Server).ServeCodec
2017/01/10 22:27:48 packer:     /usr/local/Cellar/go/1.7.3/libexec/src/net/rpc/server.go:477 +0x421
```

--- a/builder/docker/ecr_login.go
+++ b/builder/docker/ecr_login.go
@@ -52,6 +52,9 @@ func (c *AwsAccessConfig) EcrGetLogin(ecrUrl string) (string, string, error) {
 
 	exp := regexp.MustCompile("(?:http://|https://|)([0-9]*)\\.dkr\\.ecr\\.(.*)\\.amazonaws\\.com.*")
 	splitUrl := exp.FindStringSubmatch(ecrUrl)
+	if len(splitUrl) != 3 {
+		return "", "", fmt.Errorf("Failed to parse the ECR URL: %s it should be on the form <account number>.dkr.ecr.<region>.amazonaws.com", ecrUrl)
+	}
 	accountId := splitUrl[1]
 	region := splitUrl[2]
 

